challenges/cmojidra_Headstorm/cmojidraFrontEnd/frontend.php

```diff
@@ -0,0 +1,182 @@
+<!-- Submitted by Charmish Mojidra (cmojidra@andrew.cmu.edu) -->
+<!DOCTYPE html>
+<html>
+<head>
+  <link rel="icon" href="headstorm-logo.png">
+<meta name="view" content="width=device-width, initial-scale=1">
+<style>
+	@import url('https://fonts.googleapis.com/css?family=Josefin+Sans');
+
+*{
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+  outline: none;
+  font-family: 'Josefin Sans', sans-serif;
+}
+
+body{
+  background: #FF6347;
+}
+
+.wrapper{
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%,-50%);
+  max-width: 500px;
+  width: 100%;
+  background: #fff;
+  padding: 25px;
+  border-radius: 5px;
+  box-shadow: 4px 4px 2px rgba(254,236,164,1); 
+}
+
+.wrapper h2{
+  text-align: center;
+  margin-bottom: 20px;
+  text-transform: uppercase;
+  letter-spacing: 3px;
+  color: #332902;
+}
+
+.wrapper .input_field{
+  margin-bottom: 10px;
+}
+
+.wrapper .input_field input[type="text"],
+.wrapper textarea{
+  border: 1px solid #e0e0e0;
+  width: 100%;
+  padding: 10px;
+}
+
+.wrapper textarea{
+  resize: none;
+  height: 80px;
+}
+
+.wrapper .btn input[type="submit"]{
+  border: 0px;
+  margin-top: 15px;
+  padding: 10px;
+  text-align: center;
+  width: 100%;
+  background: #00CED1;
+  color: #332902;
+  text-transform: uppercase;
+  letter-spacing: 5px;
+  font-weight: bold;
+  border-radius: 25px;
+  cursor: pointer;
+}
+
+#error_message{
+  margin-bottom: 20px;
+  background: #fece0c;
+  padding: 0px;
+  text-align: center;
+  font-size: 14px;
+  transition: all 0.5s ease;
+}
+</style>
+<script src='https://www.google.com/recaptcha/api.js'></script>
+<script src="https://code.jquery.com/jquery-1.12.4.js"></script>
+<script src="https://code.jquery.com/ui/1.12.1/jquery-ui.js"></script>
+<script type="text/javascript">
+	function validate(){
+  var name = document.getElementById("name").value;
+  var subject = document.getElementById("subject").value;
+  var phone = document.getElementById("phone").value;
+  var email = document.getElementById("email").value;
+  var message = document.getElementById("message").value;
+  var country = document.getElementById("country").value;
+  var error_message = document.getElementById("error_message");
+  
+  error_message.style.padding = "10px";
+  
+  var text;
+  if(name.length < 3){
+    text = "Please Enter valid Name";
+    error_message.innerHTML = text;
+    return false;
+  }
+  if(subject.length < 5){
+    text = "Please Enter Correct Subject";
+    error_message.innerHTML = text;
+    return false;
+  }
+  if(isNaN(phone) || phone.length != 10){
+    text = "Please Enter valid Phone Number";
+    error_message.innerHTML = text;
+    return false;
+  }
+  if(email.indexOf("@") == -1 || email.length < 6){
+    text = "Please Enter valid Email";
+    error_message.innerHTML = text;
+    return false;
+  }
+  if(country.length <= 1){
```

Comment:
Would the value "@@@@@@@@@" pass this validation step? Is there another, more robust approach we can take here?

Reply:
The current validation will fail on the entered input if we pass "@@@@@@@@@". For this purpose, we have to change the code while validating and we can use the PHP functions to explode and count.

First, we will split on "@" using explode() and then if the length of the resulting array is more than 2 then the string contains more than 2 "@" which means that the email is invalid. As a second check if the length of the split array is just 2 then we will split the second part at ".". Finally, we will check if the split right part is of length 2 then the email is valid. If the length of the split right part needs to be greater than equal to 1 for the email to be valid.

For example,
cmojidra@andrew.cmu.edu

Split on @ -> cmojidra                 andrew.cmu.edu
Split Right side on "." ->   andrew       cmu         edu

$input contains entered email address


$email[] = explode("@", $input);
if(count($email) != 2)
       error_msg = "Please Enter Valid Email Address";
$temp[] = explode(".", $email[1])
if(count($temp) < 2)
        error_msg = "Please Enter Valid Email Address";
else
        print("Email is Valid");

Reply:
This approach will prevent someone from submitting "@@@@@@@@@" or something similar